_posts/2023-01-01-seraphis-jamtis-developer-opportunities.md

```diff
@@ -0,0 +1,57 @@
+
+Unfortunately there is nothing that properly documents the overall code structure, or the hierarchy of classes, or gives an overview of the functionality. This together with the fact that C++ is not the easiest of programming languages when it comes to understand a large codebase makes it a bit difficult for new devs to find their way around; but on the other hand the "culture" among the devs is very welcoming and supporting to people who are serious to contribute.
+
+You find the project [here on GitHub](https://github.com/monero-project). The most important repository is the one with the core software, e.g. the Monero *daemon* that spans up the network and manages the blockchain, see [here](https://github.com/monero-project/monero). The "place to be" for Monero devs and dev-related discussion is the Libera.chat IRC channel *#monero-dev*.
+
+### Seraphis and Jamtis
+
+A group of Monero devs started to work on two brand-new and sizeable new technologies called *Seraphis* and *Jamtis* which will improve the privacy of Monero transactions further and will solve many problems regarding current functionality. Those devs get support from the inventor of Seraphis, UkoeHB, and the inventor of Jamtis, Tevador. The resulting cryptocurrency will be quite different from today's Monero, although existing users will be offered a seamless upgrade path, and funds from all the way back to 2014 remain fully spendable indefinitely.
+
+Over the course of about one year UkoeHB built a library that implements all the cryptography behind Seraphis and e.g. contains code to create transactions. You find it [here](https://github.com/UkoeHB/monero/tree/seraphis_lib/src/seraphis).
+
+Work has now started on the so-called *Seraphis wallet* that will make heavy use of the Seraphis library. *wallet* in this context does not mean an app that end users run to manage their XMR, but that component of the Monero core code that such apps will build on. The main functionality of that component: Reading and writing wallet files that contain keys and all owned outputs, scanning the blockchain for new incoming transactions, building new transactions and submitting them to the network.
+
+The current plan is to fully replace the current component with this role called *wallet2*. This in turn will make many adjustments in the rest of the Monero core codebase necessary, and of course reworking all wallet apps to use the new component.
+
+If everything works out Seraphis and Jamtis may go into service with a hardfork that some still very rough estimates see happening in about 2 years.
```

Comment:
This could be removed as it is a repetion of https://github.com/monero-project/monero-site/pull/2119/files#diff-935e078dfa02cd893c4720a65dfa9998d664cf2518bbd175370fda911c89b8beR13

Reply:
I removed the "2 years" sentence. The technically quite important detail about wanting to replace *wallet2* is not yet mentioned in the overview and might be interesting for devs.